02_activities/assignments/assignment_2.ipynb

```diff
@@ -1,339 +1,518 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "pb9EugDCJA4c"
+   },
+   "source": [
+    "**Understanding the `check_zeros(x)` Helper Function**\n",
+    "\n",
+    "The `check_zeros(x)` function is provided as a tool to assist with your data analysis. While you do not need to modify or fully understand the internal workings of this function, it's important to grasp its input, output, and what the output signifies:\n",
+    "\n",
+    "1. **Input**:\n",
+    "   - **Parameter `x`**: This function takes an array of numbers as its input. In the context of your assignment, this array will typically represent a set of data points from your patient data, such as mean inflammation scores.\n",
+    "\n",
+    "2. **Output**:\n",
+    "   - The function returns a boolean value: either `True` or `False`.\n",
+    "\n",
+    "3. **Interpreting the Output**:\n",
+    "   - **Output is `True`**: This indicates that the array `x` contains at least one zero value. In the context of your analysis, this means that at least one patient has a mean inflammation score of 0, signaling a potential issue or anomaly in the data.\n",
+    "   - **Output is `False`**: This signifies that there are no zero values in the array `x`. For your patient data, it means no patient has a mean inflammation score of 0, and thus no apparent anomalies of this type were detected.\n",
+    "\n",
+    "**Usage in Your Analysis**:\n",
+    "When using `check_zeros(x)` in conjunction with your `patient_summary()` function in the `detect_problems()` function, you'll be checking whether any patient in your dataset has an average (mean) inflammation score of 0."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 41,
+   "metadata": {
+    "id": "_svDiRkdIwiT"
+   },
+   "outputs": [],
+   "source": [
+    "# Run this cell so you can use this helper function\n",
+    "\n",
+    "def check_zeros(x):\n",
+    "    '''\n",
+    "    Given an array, x, check whether any values in x equal 0.\n",
+    "    Return True if any values found, else returns False.\n",
+    "    '''\n",
+    "    # np.where() checks every value in x against the condition (x == 0) and returns a tuple of indices where it was True (i.e. x was 0)\n",
+    "    flag = np.where(x == 0)[0]\n",
+    "\n",
+    "    # Checks if there are any objects in flag (i.e. not empty)\n",
+    "    # If not empty, it found at least one zero so flag is True, and vice-versa.\n",
+    "    return len(flag) > 0"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 50,
+   "metadata": {
+    "id": "LEYPM5v4JT0i"
+   },
+   "outputs": [],
+   "source": [
+    "# Define your function `detect_problems` here\n",
+    "\n",
+    "def detect_problems(file_path):\n",
+    "  #YOUR CODE HERE: use patient_summary() to get the means and check_zeros() to check for zeros in the means\n",
+    "  patient_means = patient_summary(file_path, 'mean')\n",
+    "\n",
+    "  if check_zeros(patient_means >0):\n",
```

Comment:
The call to check_zeros is incorrect

Reply:
I have resubmitted it. Please let me know if it is correct now. Thank you!

Reply:
@anjali-deshpande-hub  is it good now?

Reply:
This problem is fixed.